posthog/queries/stickiness/stickiness_event_query.py

```diff
@@ -80,16 +80,18 @@ def _person_query(self):
         )
 
     def _determine_should_join_distinct_ids(self) -> None:
-        self._should_join_distinct_ids = True
+        if self._person_on_events_mode == PersonOnEventsMode.V1_ENABLED:
+            self._should_join_distinct_ids = False
+        else:
+            self._should_join_distinct_ids = True
 
     def _determine_should_join_persons(self) -> None:
         EventQuery._determine_should_join_persons(self)
         if self._person_on_events_mode != PersonOnEventsMode.DISABLED:
-            self._should_join_distinct_ids = False
             self._should_join_persons = False
 
     def aggregation_target(self):
-        return f"{self.DISTINCT_ID_TABLE_ALIAS if self._person_on_events_mode == PersonOnEventsMode.DISABLED else self.EVENT_TABLE_ALIAS}.person_id"
+        return self._person_id_alias
```

Comment:
Do these need to be here rather than on `EventQuery` itself?

Could we also add some comments explaining this or update the variable name. If I assume correctly, that means join distinct_ids if PoE disabled and join overrides if PoEv2 otherwise no joins for persons side

Reply:
And what variable name do you mean?

Reply:
`_should_join_distinct_ids`

Reply:
> I actually had deleted this but the enterprise query inherits from this and overrides it, so I reverted it as it'd be a bit of a tangential refactor

why? it looks like we're just doing the same everywhere in https://github.com/PostHog/posthog/pull/14850 it's the same for `_determine_should_join_distinct_ids` and `_determine_should_join_persons` so it looks to me like that's pretty common, perhaps instead of inheritance (which you say doesn't work?) we can use composition instead where all the person related parts are together 🤷‍♀️

Reply:
ah so not aggregation_target as above.

but still, there's a good bit of variation with `_determine_should_join_distinct_ids`. probably easy to change, but this is yet another refactor I won't be picking up now. I'm doing some minor ones along the way but will first build this in the current state. Getting to know all these queries might put me in a nice place to refactor things in a follow-up though